### DISCUSSION TITLE
Why are table subcaptions not rendering if 'library(kableExtra)' is invoked?

### DISCUSSION THREAD

**Discussion by r-scotti** [quarto-dev/quarto-cli#9586] [Q&A]
### Description

I have problems using 'tbl-subcap' chunk option.

Working on the example on page [Quarto sub-cap example](https://quarto.org/docs/authoring/cross-references.html#floats) (https://quarto.org/docs/authoring/cross-references.html#floats), all the chunks in the quarto file attached to reproduce the problem, have following lines:

```         
#| tbl-subcap: 
#|    - "first" 
#|    - "second" 
#| layout-ncol: 2 
```
Quarto file to reproduce the problem: [TestingSubCaps.pdf](https://github.com/quarto-dev/quarto-cli/files/15213587/TestingSubCaps.pdf)

Sub-captions in Table 1 are regularly produced if the 'library(kableExtra)' line is commented out.\
![image](https://github.com/quarto-dev/quarto-cli/assets/16821135/244e66c1-059d-4901-9b83-08bdaa38d31e)

I would like to have 'striped' tables, as in Table 2. 
Invoking 'kableExtra::kable_styling()' produces the effect BUT the subcaptions are lost!
![image](https://github.com/quarto-dev/quarto-cli/assets/16821135/9fcc47bb-21e0-4dd8-9792-69583e893601)

Actually just adding 'library(kableExtra)' breaks all formatting (Table 3)
![image](https://github.com/quarto-dev/quarto-cli/assets/16821135/c8d1e38c-e7b9-4c47-a31a-df623ed6e3b5)

Is this correct?

PS - Is there a way to have the chunk displayed before the table it produces?
(Possible link with previous question https://github.com/quarto-dev/quarto-cli/discussions/9546)

Many thanks for taking care.

------------
Quarto version:  1.4.553
Ubuntu version:  20.04

````qmd
---
title: "TestingSubCaps - Reproducible Quarto Document"
format: pdf
editor: visual
---

# Subcaps test

I have problems using 'tbl-subcap' chunk option.

Working on the example on page [Quarto sub-cap example](https://quarto.org/docs/authoring/cross-references.html#floats) (https://quarto.org/docs/authoring/cross-references.html#floats), all the chunks below have following lines:

```         
#| tbl-subcap: 
#|    - "first" 
#|    - "second" 
#| layout-ncol: 2 
```

Sub-captions in @tbl-tables1 are regularly produced if the 'library(kableExtra)' line is commented out.\
I would like to have 'striped' tables, as in @tbl-tables2. Invoking 'kableExtra::kable_styling()' produces the effect BUT the subcaptions are lost!

Actually just adding 'library(kableExtra)' breaks all formatting (@tbl-tables3)

Is this correct?

PS - Is there a way to have the chunk displayed before the table it produces?

```{r}
#| label: tbl-tables1
#| tbl-cap: without 'library(kableExtra)'
#| tbl-subcap:
#|  - "first"
#|  - "second"
#| layout-ncol: 2

# For 'Table 1'
library(knitr)
# library(kableExtra)

kable(head(cars))
kable(head(pressure))
```

```{r}
#| label: tbl-tables2
#| tbl-cap: "using 'kable_styling()'"
#| tbl-subcap:
#|  - "first"
#|  - "second"
#| layout-ncol: 2

# For 'Table 2'
library(knitr)

my_tbl <- function(x, ...) kable(x, ...) |> 
  kableExtra::kable_styling(latex_options = "striped")

head(cars) |> my_tbl()
head(pressure) |> my_tbl()
```

```{r}
#| label: tbl-tables3
#| tbl-cap: adding just 'library(kableExtra)'
#| tbl-subcap:
#|  - "first"
#|  - "second"
#| layout-ncol: 3

# For 'Table 2'
library(knitr)
library(kableExtra)

kable(head(cars))
kable(head(pressure))
```
````

### $ quarto check all
Quarto 1.4.553
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.553
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/roberto/.TinyTeX/bin/x86_64-linux
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.12 (Conda)
      Path: /node/local/TOOLS/anaconda3/bin/python
      Jupyter: 4.9.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /usr/lib/R
      LibPaths:
        - /home/roberto/R/x86_64-pc-linux-gnu-library/4.3
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
$

**Comment by mcanouil** [comment]
Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

This being said:

- https://github.com/quarto-dev/quarto-cli/issues/4517

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by r-scotti** [reply]
I edited copying at the end the linked file in the code chunk  and adding the output of quarto check. 
Did I correctly reply to the requests?
#4517 already noticed a problem with KableExtra but the question is still open, isn't it?

**Reply by cderv** [reply]
I commented on #4517 about this. 

This is basically a current issue with LaTeX output tables from `kableExtra` 

> Sub-captions in Table 1 are regularly produced if the 'library(kableExtra)' line is commented out.

Check `keep-md: true` but I believe here tables will be Markdown table in the results of the cell evaluation 

> Invoking 'kableExtra::kable_styling()' produces the effect BUT the subcaptions are lost!

This will produce LaTeX code, but then #4517 is hit. 

> Actually just adding 'library(kableExtra)' breaks all formatting (Table 3)

**kableExtra** is a library that choose to output LaTeX formatted table in output when `pdf` is the output format type. This means LaTeX code will be produced. So they won't get the same formatting as Markdown table parsed by Pandoc and transformed to LaTeX by pandoc. This is probably because **kableExtra** does not use same table package and option as Pandoc will. 

Hope it clarifies
